chain/syncer_integration_test.go

```diff
@@ -251,6 +251,7 @@ func initSyncTest(t *testing.T, con consensus.Protocol, genFunc func(cst *hamt.C
 	chainStore := chain.NewStore(chainDS, cst, &state.TreeStateLoader{}, calcGenBlk.Cid())
 
 	fetcher := th.NewTestFetcher()
+	fetcher.AddSourceBlocks(calcGenBlk)
```

Comment:
The genesis block needs to exist in the store the fetcher pulls from, this is because the fetcher will stop fetching when it finds a block it has seen before and this can sometimes be the genesis block.

Reply:
It doesn't make sense that the fetch is ever asked for the genesis block, `HasTipSetAndState` should always return true and hence it doesn't need to be fetched. Maybe my suggested change to the predicate above will fix this.

Reply:
Your suggestion on fetching parents fixes this -- thanks!